kubectl crashing when --validate used

```
$ cluster/kubectl.sh create --validate -f examples/guestbook/redis-master-service.json
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x505433]

goroutine 1 [running]:
github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation.(*SwaggerSchema).ValidateObject(0xc2080920b0, 0x8944e0, 0xc2080ca180, 0xc208147fb8, 0x7, 0x0, 0x0, 0xc2081c0070, 0xe, 0x0, ...)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation/schema.go:101 +0x8f3
github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation.(*SwaggerSchema).ValidateBytes(0xc2080920b0, 0xc20809a180, 0xfa, 0x174, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/api/validation/schema.go:79 +0x36b
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd.(*clientSwaggerSchema).ValidateBytes(0xc2080e94a0, 0xc20809a180, 0xfa, 0x174, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/cmd.go:403 +0x334
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd.func·018(0xc2080f1da0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/create.go:93 +0x134
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.func·004(0xc2080f1da0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:320 +0xe8
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.func·005(0xc2080f1da0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:348 +0xfa
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.func·002(0xc2080f1da0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:172 +0x5b
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.(*PathVisitor).Visit(0xc2080b0240, 0xc2080b0340, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:206 +0x412
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.EagerVisitorList.Visit(0xc20802b960, 0x1, 0x1, 0xc2080b0300, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:176 +0x17e
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.(*EagerVisitorList).Visit(0xc2080b0260, 0xc2080b0300, 0x0, 0x0)
	<autogenerated>:77 +0xbd
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.FlattenListVisitor.Visit(0x7f7d194871a8, 0xc2080b0260, 0xc2080cb4d0, 0xc2080b02c0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:363 +0x10b
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.(*FlattenListVisitor).Visit(0xc2080b0280, 0xc2080b02c0, 0x0, 0x0)
	<autogenerated>:86 +0xbd
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.DecoratedVisitor.Visit(0x7f7d194871d0, 0xc2080b0280, 0xc20802b990, 0x2, 0x2, 0xc2080b02a0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/visitor.go:321 +0x110
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.(*DecoratedVisitor).Visit(0xc2080cb500, 0xc2080b02a0, 0x0, 0x0)
	<autogenerated>:97 +0xbc
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource.(*Result).Visit(0xc208082850, 0xc2080b02a0, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource/result.go:77 +0x73
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd.RunCreate(0xc2080828c0, 0x7f7d19484c50, 0xc208032008, 0xc20805c780, 0xc20802b2b0, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/create.go:104 +0x53d
github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd.func·017(0xc20805c780, 0xc20803a7c0, 0x0, 0x4)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/create.go:50 +0xc8
github.com/spf13/cobra.(*Command).execute(0xc20805c780, 0xc20803a6c0, 0x4, 0x4, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:444 +0x410
github.com/spf13/cobra.(*Command).findAndExecute(0xc20805c280, 0xc20800a010, 0x5, 0x5, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:408 +0xd4
github.com/spf13/cobra.(*Command).Execute(0xc20805c280, 0x0, 0x0)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/spf13/cobra/command.go:497 +0xb1b
main.main()
	/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/cmd/kubectl/kubectl.go:29 +0x104

goroutine 5 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0xfb2900)
	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x78
created by github.com/golang/glog.init·1
	/go/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2a7

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/src/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 6 [syscall]:
os/signal.loop()
	/usr/src/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/usr/src/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 9 [IO wait]:
net.(*pollDesc).Wait(0xc208083b10, 0x72, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208083b10, 0x0, 0x0)
	/usr/src/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc208083ab0, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x7f7d19484f00, 0xc2081aabf0)
	/usr/src/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc208032288, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/src/go/src/net/net.go:121 +0xdc
net/http.noteEOFReader.Read(0x7f7d19488a78, 0xc208032288, 0xc208092318, 0xc20800f000, 0x1000, 0x1000, 0x9225a0, 0x0, 0x0)
	/usr/src/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc2080b15c0, 0xc20800f000, 0x1000, 0x1000, 0xc208012000, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc2080fe420)
	/usr/src/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc2080fe420, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/src/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2080922c0)
	/usr/src/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:660 +0xc9f

goroutine 10 [select]:
net/http.(*persistConn).writeLoop(0xc2080922c0)
	/usr/src/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/usr/src/go/src/net/http/transport.go:661 +0xcbc
```

--- a/pkg/api/validation/schema.go
+++ b/pkg/api/validation/schema.go
@@ -98,6 +98,10 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, apiVersion, fieldName, t
 			glog.V(2).Infof("couldn't find properties for %s, skipping", key)
 			continue
 		}
+		if details.Type == nil {
+			glog.V(2).Infof("nil details for %s, skipping", key)
+			continue
+		}
 		fieldType := *details.Type
 		if value == nil {
 			glog.V(2).Infof("Skipping nil field: %s", key)

